src/azure-cli/requirements.txt

```diff
@@ -0,0 +1,167 @@
+azure-cli-botservice==0.1.4
+azure-cli-cdn==0.2.0
+azure-cli-cloud==2.1.0
+azure-cli-cognitiveservices==0.2.4
+azure-cli-command-modules-nspkg==2.0.2
+azure-cli-configure==2.0.20
+azure-cli-consumption==0.4.2
+azure-cli-container==0.3.11
+azure-cli-core==2.0.55
+azure-cli-cosmosdb==0.2.7
+azure-cli-dla==0.2.3
+azure-cli-dls==0.1.7
+azure-cli-dms==0.1.1
+azure-cli-eventgrid==0.2.0
+azure-cli-eventhubs==0.3.2
+azure-cli-extension==0.2.3
+azure-cli-feedback==2.1.4
+azure-cli-find==0.2.13
+azure-cli-hdinsight==0.2.0
+azure-cli-interactive==0.4.1
+azure-cli-iot==0.3.4
+azure-cli-iotcentral==0.1.5
+azure-cli-keyvault==2.2.9
+azure-cli-lab==0.1.5
+azure-cli-maps==0.3.3
+azure-cli-monitor==0.2.8
+azure-cli-network==2.3.0
+azure-cli-nspkg==3.0.3
+azure-cli-policyinsights==0.1.0
+azure-cli-profile==2.1.2
+azure-cli-rdbms==0.3.5
+azure-cli-redis==0.3.2
+azure-cli-relay==0.1.2
+azure-cli-reservations==0.4.1
+azure-cli-resource==2.1.8
+azure-cli-role==2.3.0
+azure-cli-search==0.1.1
+azure-cli-servicebus==0.3.2
+azure-cli-servicefabric==0.1.11
+azure-cli-signalr==1.0.0
+azure-cli-sql==2.1.7
+azure-cli-storage==2.3.0
+azure-cli-telemetry==1.0.0
+azure-cli-vm==2.2.12
+azure-common==1.1.17
+azure-datalake-store==0.0.39
+azure-graphrbac==0.53.0
+azure-keyvault==1.1.0
+azure-mgmt-advisor==2.0.1
+azure-mgmt-authorization==0.50.0
+azure-mgmt-batch==5.0.1
+azure-mgmt-batchai==2.0.0
+azure-mgmt-billing==0.2.0
+azure-mgmt-botservice==0.1.0
+azure-mgmt-cdn==3.0.0
+azure-mgmt-cognitiveservices==3.0.0
+azure-mgmt-compute==4.4.0
+azure-mgmt-consumption==2.0.0
+azure-mgmt-containerinstance==1.2.1
+azure-mgmt-containerregistry==2.6.0
+azure-mgmt-containerservice==4.2.2
+azure-mgmt-cosmosdb==0.5.2
+azure-mgmt-datalake-analytics==0.2.0
+azure-mgmt-datalake-nspkg==3.0.1
+azure-mgmt-datalake-store==0.5.0
+azure-mgmt-datamigration==0.1.0
+azure-mgmt-devtestlabs==2.2.0
+azure-mgmt-dns==2.1.0
+azure-mgmt-eventgrid==0.4.0
+azure-mgmt-eventhub==2.2.0
+azure-mgmt-hdinsight==0.2.0
+azure-mgmt-iotcentral==1.0.0
+azure-mgmt-iothub==0.6.0
+azure-mgmt-iothubprovisioningservices==0.2.0
+azure-mgmt-keyvault==1.1.0
+azure-mgmt-loganalytics==0.2.0
+azure-mgmt-managementgroups==0.1.0
+azure-mgmt-maps==0.1.0
+azure-mgmt-marketplaceordering==0.1.0
+azure-mgmt-media==1.0.1
+azure-mgmt-monitor==0.5.2
+azure-mgmt-msi==0.2.0
+azure-mgmt-network==2.4.0
+azure-mgmt-nspkg==3.0.2
+azure-mgmt-policyinsights==0.1.0
+azure-mgmt-rdbms==1.5.0
+azure-mgmt-recoveryservices==0.1.0
+azure-mgmt-recoveryservicesbackup==0.1.1
+azure-mgmt-redis==5.0.0
+azure-mgmt-relay==0.1.0
+azure-mgmt-reservations==0.3.1
+azure-mgmt-resource==2.0.0
+azure-mgmt-search==2.0.0
+azure-mgmt-servicebus==0.5.3
+azure-mgmt-servicefabric==0.2.0
+azure-mgmt-signalr==0.1.1
+azure-mgmt-sql==0.11.0
+azure-mgmt-storage==3.1.1
+azure-mgmt-trafficmanager==0.51.0
+azure-mgmt-web==0.40.0
+azure-multiapi-storage==0.2.2
+azure-nspkg==3.0.2
+azure-storage-blob==1.3.1
+azure-storage-common==1.4.0
+azure-storage-nspkg==3.1.0
+bcrypt==3.1.6
```

Comment:
So these dependencies we don't need to lock down per se, just the CLI modules. At the same time, will a generic `pip install azure-cli==2.0.55` honor a requirements file, or do these declarations need to be moved into `azure-cli`'s `setup.py` file?

Reply:
Applications should always be installed in a venv. azure-cli should pin everything "azure" specific. The "requirements.txt" should pin everything by design.
Should we pin everything in the setup.py of azure-cli? I tend to say yes, but the question could be discussed.

Reply:
> Should we pin everything in the setup.py of azure-cli? I tend to say yes...

Help me understand the difference here, @lmazuel. If we're already shipping a `requirements.txt` for each of our target environments, isn't `setup.py` just another place we have to edit version numbers each release?

Reply:
https://caremad.io/posts/2013/07/setup-vs-requirement/

Reply:
This article is part of the journey of every Python developer :)